include/polynomial.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <cmath>
+#include <array>
+#include <vector>
+
+// Implimented according to https://en.wikipedia.org/wiki/Cubic_equation
+
+namespace quda
+{
+
+  constexpr double PI = 3.141592653589793;
```

Comment:
Can just use `M_PI` (provided by the math library)

Reply:
`M_PI` does seem to be present in all compilers? I did a search and I do not find it in the C++ standard.

Reply:
I learned something new here, that `M_PI` is not part of math.h / cmath, and is compiler dependent.  It's definitely supported in all compilers that QUDA is used by, and we do utilize `M_PI` throughout QUDA.

If one is concernedd with using `M_PI`, we could add the following to `quda_constant.h`
```c++
#ifndef M_PI
#define M_PI (3.14159265358979323846264338327950288)
#endif /* M_PI */
```

Reply:
Personally I am fine with using `M_PI` directly but I think the above solution is better. :)

Reply:
https://github.com/lattice/quda/pull/1061/commits/d776b593adac861010d314b14523ba261ada3c14